arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArLocationDataSource.kt

```diff
@@ -529,14 +529,23 @@ private fun createCalendarFromTimeInMillis(timeInMillis: Long): Calendar {
  * @since 100.6.0
  */
 private fun android.location.Location.toEsriLocation(lastKnown: Boolean): LocationDataSource.Location {
-    val position = Point(longitude, latitude, SpatialReference.create(4326))
-    var verticalAccuracy = java.lang.Double.NaN
-    val timeStamp = createCalendarFromTimeInMillis(time)
-
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-        verticalAccuracy = verticalAccuracyMeters.toDouble()
+    // Get vertical accuracy when available
+    val verticalAccuracy =
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) verticalAccuracyMeters.toDouble() else Double.NaN
+
+    // If vertical accuracy is greater than 0, provide a position with Z value
+    val position = if (verticalAccuracy.isNaN().not() && verticalAccuracy >= 0) Point(
```

Comment:
Is it correct to check `>=`? The doc for `getVerticalAccuracyMeters()` says:

> If this location does not have a vertical accuracy, then 0.0 is returned.

Also we're ignoring the altitude if `Build.VERSION.SDK_INT < Build.VERSION_CODES.O`. I wonder if we should use `hasAltitude` here rather than `verticalAccuracy`. I see in https://devtopia.esri.com/runtime/queen-annes-revenge/issues/198#issuecomment-1817662 that iOS uses `verticalAccuracy` when it decides whether or not to use the altitude, but maybe iOS doesn't have a `hasAltitude` property to consult? (just guessing about that)